File: tests/fixtures/real_world/ruby/state/file_lifecycle.rb
def read_and_leak(path)
  f = File.open(path, 'r')
  data = f.read
  data
end

def read_and_close(path)
  f = File.open(path, 'r')
  data = f.read
  f.close
  data
end

def double_close(path)
  f = File.open(path, 'r')
  f.close
  f.close
end

def use_after_close(path)
  f = File.open(path, 'r')
  f.close
  data = f.read
  data
end

File: tests/fixtures/real_world/python/state/file_lifecycle.py
def read_and_leak(path):
    f = open(path, 'r')
    data = f.read()
    return data

def read_and_close(path):
    f = open(path, 'r')
    data = f.read()
    f.close()
    return data

def double_close(path):
    f = open(path, 'r')
    f.close()
    f.close()

def use_after_close(path):
    f = open(path, 'r')
    f.close()
    data = f.read()
    return data
